docs/public/specifications/tests/Basic-TP/basic01.md

```diff
@@ -89,169 +89,167 @@ DNS queries follow, unless otherwise specified below, what is specified for
+       SOA and name server IP *Server Address* and go to next server in
+       *Remaining Servers* if one or more of the following matches:
+          * No DNS response.
+          * [RCODE Name] besides NoError in response.
+          * AA bit not set in response.
+          * No SOA record in answer section
+          * Owner name of SOA record is not *Zone Name*.
+   6.  [Send] *Zone Name NS Query* to *Server Address*.
+   7.  Output [B01_SERVER_ZONE_ERROR] with query name *Zone Name*, [query type]
+       SOA and name server IP *Server Address* and go to next server in
+       *Remaining Servers* if one or more of the following matches:
+          * No DNS response.
+          * [RCODE Name] besides NoError in response.
+          * AA bit not set in response.
+          * No NS records in answer section
+          * Owner name of NS records is not *Zone Name*.
+   8.  Extract the name server names from the NS records and any address records
+       in the additional section.
+   9.  Do [DNS Lookup] of name server names (A and AAAA) not already listed in the
+       additional section of the response.
+   10. For each IP address add the IP address and *Zone Name* to the
+       *Remaining Servers* set unless the IP address is already listed in
+       *Handled Servers* together with *Zone Name*.
+   11. Create "Intermediate Query Name" by copying *Zone name* as start value.
+   12. Run a loop processing *Server Address* (jumps back here from the steps
+       below).
+       1. Extend "Intermediate Query Name" by adding one more label to the left
+          by copying the equivalent label from *Child Zone*. (See "Example 1"
+          below.)
+       2. Create a [DNS queries][DNS Query] with query name
+          *Intermediate Query Name* and [query type] SOA
+          ("Intermediate SOA query").
+       3. [Send] *Intermediate SOA Query* to *Server Address*. (See "Example 2"
+          below.)
+       4. Output [B01_SERVER_ZONE_ERROR] with query name *Intermediate Query Name*
+          and [query type] SOA and name server IP *Server Address* and go to next
+          server in *Remaining Servers* if there is no DNS response.
+       5. If the response has an SOA record with owner name
+          *Intermediate Query Name* in the answer section, with the AA bit
+          set and [RCODE Name] NoError then do:
+          1. If *Intermediate Query Name* is equal to *Child Zone* then
+             1. Save *Server Address* and *Zone Name* to the *Parent Found* set
+                and to the *AA SOA Found* set.
+             2. Go to next server in *Remaining Servers*.
+          2. Else do:
+             1. Create a [DNS queries][DNS Query] with query name
+                *Intermediate Query Name* and [query type] NS
+                ("Intermediate NS query").
+             2. [Send] *Intermediate NS Query* to *Server Address*.
+             3. Output [B01_SERVER_ZONE_ERROR] with query name
```

Comment:
My remarks and suggestions for step 7 also apply here.

Reply:
I am not sure what you refer to.

Reply:
Already fixed.